File: Python/Programmars/Level_2/짝지어 제거하기.py
from collections import Counter


def solution(s):
    if len(s) % 2 != 0:
        return 0
    if any(x % 2 != 0 for x in Counter(s).values()):
        return 0
    stack = []
    stack.append(s[0])
    for i in range(1, len(s)):
        if stack and stack[-1] == s[i]:
            stack.pop()
        else:
            stack.append(s[i])
    if stack:
        return 0
    else:
        return 1


print(solution('baabaa'))  # 1
print(solution('cdcd'))  # 0

File: JavaScript/Programmars/Level_2/짝지어 제거하기.js
function solution(s) {
  if (s.length % 2 === 1) return 0;

  const stack = [];
  for (let char of s) {
    if (stack[stack.length - 1] === char) {
      stack.pop();
    } else {
      stack.push(char);
    }
  }

  return stack.length === 0 ? 1 : 0;
}

console.log(solution('baabaa')); // 1
console.log(solution('cdcd')); // 0
